File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/enterprise/contacts/ui/EnterpriseContactsUI.java
package com.fintechplatform.ui.enterprise.contacts.ui;

import android.content.Context;
import android.content.Intent;

import com.android.volley.toolbox.Volley;
import com.fintechplatform.api.enterprise.api.EnterpriseAPIModule;
import com.fintechplatform.api.net.NetData;
import com.fintechplatform.api.net.NetModule;
import com.fintechplatform.ui.models.DataAccount;

public class EnterpriseContactsUI {
    private String hostName;
    private DataAccount configuration;

    static EnterpriseContactsUI instance;

    public EnterpriseContactsUI(String hostName, DataAccount configuration) {
        this.hostName = hostName;
        this.configuration = configuration;
    }

    private EnterpriseContactsViewComponent buildContactsViewComponet(Context context, EnterpriseContactsContract.View view) {
        return DaggerEnterpriseContactsViewComponent.builder()
                .netModule(new NetModule(new NetData(Volley.newRequestQueue(context), instance.hostName)))
                .enterpriseContactsPresenterModule(new EnterpriseContactsPresenterModule(view, instance.configuration))
                .enterpriseAPIModule(new EnterpriseAPIModule(instance.hostName))
                .build();
    }
    public EnterpriseContactsViewComponent createContactsComponent(Context context, EnterpriseContactsContract.View view) {
        return instance.buildContactsViewComponet(context, view);
    }

    public void start(Context context) {
        instance = this;
        Intent intent = new Intent(context, EnterpriseContactsActivity.class);
        context.startActivity(intent);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/enterprise/contacts/EnterpriseContactsUI.kt
package com.fintechplatform.ui.enterprise.contacts

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.android.volley.toolbox.Volley
import com.fintechplatform.api.enterprise.api.EnterpriseAPIModule
import com.fintechplatform.api.net.NetModule
import com.fintechplatform.ui.enterprise.contacts.di.DaggerEnterpriseContactsViewComponent
import com.fintechplatform.ui.enterprise.contacts.di.EnterpriseContactsPresenterModule
import com.fintechplatform.ui.enterprise.contacts.di.EnterpriseContactsViewComponent
import com.fintechplatform.ui.models.DataAccount

class EnterpriseContactsUI(private val hostName: String, private val configuration: DataAccount) {

    fun startActivity(context: Context) {
        val intent = Intent(context, EnterpriseContactsActivity::class.java)
        val args = Bundle()
        args.putString("hostname", hostName)
        args.putParcelable("dataAccount", configuration)
        intent.putExtras(args)
        context.startActivity(intent)
    }

    fun createFragment(): EnterpriseContactsFragment {
        return EnterpriseContactsFragment.newInstance(hostName, configuration)
    }

    object Builder {
        fun buildContactsViewComponet(context: Context, view: EnterpriseContactsContract.View, hostName: String, dataAccount: DataAccount): EnterpriseContactsViewComponent {
            return DaggerEnterpriseContactsViewComponent.builder()
                    .netModule(NetModule(Volley.newRequestQueue(context), hostName))
                    .enterpriseContactsPresenterModule(EnterpriseContactsPresenterModule(view, dataAccount))
                    .enterpriseAPIModule(EnterpriseAPIModule(hostName))
                    .build()
        }
    }
}